src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/SecretUtils.java

```diff
@@ -200,11 +201,13 @@ public static String getNonNullSecretData(Secret s, String key, String exception
      * @return the custom mapping for the key or {@code key} (identical object) if there is no custom mapping.
      */
     public static String getKeyName(Secret s, String key) {
-        String customMapping = s.getMetadata().getAnnotations().get(JENKINS_IO_CREDENTIALS_KEYBINDING_ANNOTATION_PREFIX + key);
-        if (customMapping != null && customMapping.length() > 0) {
-            return customMapping;
+        Map<String, String> annotations = s.getMetadata().getAnnotations();
+        if (annotations != null && annotations.size() > 0){
+            String customMapping = s.getMetadata().getAnnotations().get(JENKINS_IO_CREDENTIALS_KEYBINDING_ANNOTATION_PREFIX + key);
```

Comment:
Should use the `annotations` variable just introduced. Also there is no need to check the size:

```java
if (annotations != null) {
    String customMapping = annotations.get(JENKINS_IO_CREDENTIALS_KEYBINDING_ANNOTATION_PREFIX + key);
```